eduVPN/ProviderService.swift

```diff
@@ -551,10 +552,22 @@ class ProviderService {
             request.setValue("Bearer \(accessToken)", forHTTPHeaderField: "Authorization")
             
             let task = self.urlSession.dataTask(with: request) { (data, response, error) in
-                guard let data = data, let response = response as? HTTPURLResponse, 200..<300 ~= response.statusCode else {
+                guard let data = data, let response = response as? HTTPURLResponse else {
+                    handler(.failure(error ?? Error.unknown))
+                    return
+                }
```

Comment:
This block and the `200..<300` guard statement seem to be repeated a couple of times. Maybe it's possible to de-duplicate this by lifting it into a network service, or something similar?

Reply:
I've looked into that, but due to differences in the type of the handler it gets rather ugly.